lib/nimble_pool.ex

```diff
@@ -441,7 +443,7 @@ defmodule NimblePool do
         {:noreply, state}
 
       %{} ->
-        exit(:unexpected_remove)
+        {:noreply, state}
```

Comment:
I think we can add the `exit` back because we no longer send the cancel_ref on timeout?